Harmonizing Congo DR hyphen, spaces, initcap in global hd

nihaas21 opened the pull request:
Note: The aggregation of regions still needs to be dealt with, this is just to keep the format similar to what we edited in Boost data.

Commit:
Harmonizing Congo DR hyphen, spaces, initcap in global hd

Note: The aggregation of regions still needs to be dealt with, this is just to keep the format similar to what we edited in Boost data.

weilu commented on human_development/global_data_lab_hid_transform_load_dlt.py:
```diff
@@ -73,9 +73,12 @@ def global_data_lab_hd_index():
                 ).otherwise(
                     F.col("Region")
                 )
+            ).when(F.col("country_name") == 'Congo Democratic Republic',
```
This might need to be aligned with DRC name used in BOOST: "Congo, Dem. Rep."

Commit:
Harmonizing Congo Dem Rep name

weilu commented on human_development/global_data_lab_hid_transform_load_dlt.py:
```diff
@@ -73,9 +73,14 @@ def global_data_lab_hd_index():
                 ).otherwise(
                     F.col("Region")
                 )
+            ).when(F.col("country_name") == 'Congo Democratic Republic',
+                F.lit("Congo, Dem. Rep.")    
+            ).when(F.col("country_name") == 'Congo, Dem. Rep.',
+                F.initcap(F.regexp_replace(F.col("Region"), "-", " "))   
```
@nihaas21 I'm confused – are there two possible names for DRC that you are handling here?

nihaas21 replied:
The name of DRC in global_hd_index is Congo Democratic Republic whereas in Boost it is Congo, Dem. Rep.
Aligning that and also removing hyphens from the regions associated.

weilu replied:
This doesn't change the `country_name` stored in the database as it is modifying the `adm1_name` column. Please remove unneccessary code here and make the changes where it should be made.

Commit:
Removing parentheses from Paraguay and Tunisia regions

Please note that changes made in the last lines of code, for adjusting aliases were not being pulled from main so had to manually copy that (in case it points out to some merge conflicts later)

weilu commented on human_development/global_data_lab_hid_transform_load_dlt.py:
```diff
@@ -73,15 +73,45 @@ def global_data_lab_hd_index():
+                ).when(F.col("Region") == 'North-East (Caaguazu, Alto Parana, Canideyu)',
+                    F.lit("North-East")
+                ).otherwise(
+                F.col("Region"))
```
The pattern here is that we want to remove the last parathesis with any content inside then be rid of the trailing space. This can be more succinctly achieved through the use of a regex expression:

```
F.regexp_replace(F.col("Region"), r'\s*\([^)]*\)$', "")
```

weilu reviewed (changes requested):
@nihaas21 the code from the previous commits on DRC region name harmonization seems to have disappeared. Can you check?

Commit:
Removing parentheses from Tunisia and Paraguay GDL

weilu commented:
@nihaas21 the regex approach I suggested above didn't work?

weilu commented:
Code in existing repo on line 12 `.withColumn("Region", F.trim(F.regexp_replace(F.col("Region"), "\\(.*\\)", "")))` already removes parenthesis with content inside. Ran code on main branch and verified the data does not contain parenthesis for the countries in question:

country_name	adm1_name	count(1)
Congo, Dem. Rep.	Bandundu	33
Congo, Dem. Rep.	Bas-Congo	33
Congo, Dem. Rep.	Equateur	33
Congo, Dem. Rep.	Kasai Occidental	33
Congo, Dem. Rep.	Kasai Oriental	33
Congo, Dem. Rep.	Katanga	33
Congo, Dem. Rep.	Kinshasa	33
Congo, Dem. Rep.	Maniema	33
Congo, Dem. Rep.	Nord-Kivu	33
Congo, Dem. Rep.	Orientale	33
Congo, Dem. Rep.	Sud-Kivu	33
Congo, Dem. Rep.	Total	33
Paraguay	Central	33
Paraguay	North-East	33
Paraguay	North-West	33
Paraguay	South-East	33
Paraguay	South-West	33
Paraguay	Total	33
Tunisia	Centre Est	33
Tunisia	Centre Ouest	33
Tunisia	Grand Tunis	33
Tunisia	Nord Est	33
Tunisia	Nord Ouest	33
Tunisia	Sud Est	33
Tunisia	Sud Ouest	33
Tunisia	Total	33
